Certification/Loader.php

```diff
@@ -21,62 +21,108 @@
  */
 class Loader
 {
+
     /**
-     * @var integer
+     * Total questions count
+     * @var int
      */
-    static public $count;
+    private static $count = null;
 
     /**
      * Returns a new set of randomized questions
      *
      * @param integer $number
-     * @param array   $categories
+     * @param array $categories
      *
      * @return Set
      */
-    static public function init($number, array $categories, $path)
+    public static function init($number, array $categories, $path)
     {
+        /** @var array $data */
         $data = self::prepareFromYaml($categories, $path);
+        self::$count = count($data);
+        shuffle($data);
+        $questions = self::mapQuestions(array_slice($data, 0, $number));
 
-        if (!$data) {
-            return new Set(array());
-        }
-
-        $dataMax = count($data) - 1;
-
-        $questions = array();
-
-        for ($i = 0; $i < $number; $i++) {
-            do {
-                $random = rand(0, $dataMax);
-            } while (isset($questions[$random]) && count($questions) < $dataMax);
-
-            $item = $data[$random];
-
-            $answers = array();
-
-            foreach ($item['answers'] as $dataAnswer) {
-                $answers[] = new Answer($dataAnswer['value'], $dataAnswer['correct']);
-            }
+        return new Set($questions);
+    }
 
-            if (!isset($item['shuffle']) || true === $item['shuffle']) {
-                shuffle($answers);
-            }
+    /**
+     * Counts total of available questions
+     *
+     * @return integer
+     */
+    public static function count($path = null)
+    {
+        if (!is_null(self::$count)) {
+            return self::$count;
+        } elseif ($path) {
+            return count(self::prepareFromYaml([], $path));
+        } else {
+            throw new \ErrorException('Provide $path to config file');
+        }
+    }
 
-            $questions[$random] = new Question($item['question'], $item['category'], $answers);
+     /**
+     * Get list of all categories
+     *
+     * @return array
+     */
+    public static function getCategories($path)
+    {
+        $categories = array();
+        $files = self::prepareFromYaml(array(), $path);
+        foreach ($files as $file) {
+            $categories[] = $file['category'];
         }
 
-        return new Set($questions);
+        return array_unique($categories);
```

Comment:
It could be better to re-index the array by calling `array_values` before returning it.